SC_redesign/Crop_and_Soil/soil/soil_erosion.py

```diff
@@ -0,0 +1,251 @@
+                                                        support_practice_factor, topographic_factor, fragment_factor)
+        self.data.eroded_sediment += self._determine_adjusted_sediment_yield(sediment_yield,
+                                                                             self.data.snow_cover_water_content)
+
+    # --- Static methods ---
+    @staticmethod
+    def _determine_coarse_sand_factor(percent_sand_content: float, percent_silt_content: float) -> float:
+        """calculates factor based on sand content for use in determining soil erodibility factor
+
+        Args:
+            percent_sand_content: percent of soil content that is sand
+            percent_silt_content: percent of soil content that is silt
+
+        Returns:
+            factor based on sand content (unitless)
+
+        SWAT Reference: 4:1.1.6
+        """
+        return 0.2 + 0.3 * exp((-0.256) * percent_sand_content * (1 - (percent_silt_content / 100)))
+
+    @staticmethod
+    def _determine_clay_silt_ratio_factor(percent_silt_content: float, percent_clay_content: float) -> float:
+        """calculates factor based on the clay-silt ratio for use in calculating soil erodibility factor
+
+        Args:
+            percent_silt_content: percent of silt in the given layer of soil
+            percent_clay_content: percent of clay in the given layer of soil
+
+        Returns:
+            clay-silt ratio factor (unitless)
+
+        SWAT Reference: 4:1.1.7
+        """
+        if percent_silt_content == 0 and percent_clay_content == 0:
+            raise ValueError("Cannot have percent silt content and percent clay content both be 0")
+        return (percent_silt_content / (percent_clay_content + percent_silt_content)) ** 0.3
+
+    @staticmethod
+    def _determine_carbon_content_factor(percent_organic_carbon: float) -> float:
+        """calculates factor based on percent of organic carbon content for use in calculating soil erodibility factor
+
+        Args:
+            percent_organic_carbon: percent of organic carbon content in the given layer of soil
+
+        Returns:
+            organic carbon percent factor in the given layer of soil
+
+        SWAT Reference: 4:1.1.8
+        """
+        return 1 - ((0.25 * percent_organic_carbon) / (percent_organic_carbon +
+                                                       exp(3.72 - (2.95 * percent_organic_carbon))))
+
+    @staticmethod
+    def _determine_high_sand_factor(percent_sand_content: float) -> float:
+        """calculates factor based on percent sand content for use in calculating soil erodibility factor
+
+        Args:
+            percent_sand_content: percent of sand in the given layer of soil
+
+        Returns:
+            factor for adjusting soil erodibility factor based on high sand contents
+
+        SWAT Reference: 4:1.1.9
+        """
+        inverse_sand_percentage = 1 - (percent_sand_content / 100)
+        # TODO: better name for this variable
```

Comment:
This isn't important at al, but how about

```suggestion
        not_sand_fraction = 1 - (percent_sand_content / 100)
```

Reply:
I like it.